src/wallet/wallet2.cpp

```diff
@@ -3110,11 +3114,10 @@ void wallet2::pull_hashes(uint64_t start_height, uint64_t &blocks_start_height,
 //----------------------------------------------------------------------------------------------------
 void wallet2::process_parsed_blocks(uint64_t start_height, const std::vector<cryptonote::block_complete_entry> &blocks, const std::vector<parsed_block> &parsed_blocks, uint64_t& blocks_added, std::map<std::pair<uint64_t, uint64_t>, size_t> *output_tracker_cache)
 {
-  size_t current_index = start_height;
```

Comment:
I understand why you moved `current_index` to where it's used the first time, but this is an unrelated change for this PR. As a result, half of the changes in this function are unnecessary.

> Patches should be self contained. A good rule of thumb is to have one patch per separate issue, feature, or logical change. Also, no other changes, such as random whitespace changes, reindentation, or fixing typos, spelling, or wording, unless user visible. Following the code style of the particular chunk of code you're modifying is encouraged. Proper squashing should be done (eg, if you're making a buggy patch, then a later patch to fix the bug, both patches should be merged).

Reply:
I believe this change is necessary to prevent future issues. I had to choose between two variables, neither of which was correct for my PR needs. `is_in_bounds(start_height)` must be called before `start_height` is used, but it was accessed via `current_index`, which is mutated later in the function. This mix-up could lead to future issues.

Such a trivial change is easy to review and quick to fix in any merge conflict. Leaving it as is would be incorrect, because the current code already mixes mutable and immutable variables. Probably reviewing the commit history would show that this mix-up comes from previous merges that made minimal changes without considering the context.

I can split the commit in two to clarify the change if necessary, but creating another PR only for this seems excessive.

Reply:
Splitting into two commits is fine. And while you're at it, `start_height` should then be declared as `const uint64_t` in function definition and declaration, if it's not supposed to be changed in the function.

Reply:
Sure, I've split into two commits and declared `start_height` as const.